tapa/common/graph.py

```diff
@@ -99,6 +99,12 @@ def get_floorplan_slot(
         """Return the task grouping the tasks floorplanned in the given slot."""
         # Construct obj of the slot by modifying the top task
         new_obj = self.get_top_task_def().to_dict()
+
+        assert isinstance(new_obj["code"], str)
+        new_obj["code"] = new_obj["code"].replace(
+            self.get_top_task_name(), slot_name, -1
+        )
```

Comment:
Consider removing the `-1` argument. It is unnecessary, and does not convey meaningful information for readers who are not familiar with `str.replace`.

Reply:
Also, do we depend on the correctness of the generated `code`? Plain string substitution without parsing C++ can lead to incorrect results.